client/src/controllers/DrilldownController.test.js

```diff
@@ -0,0 +1,214 @@
+import { Application } from '@hotwired/stimulus';
+import { DrilldownController } from './DrilldownController';
+import { DropdownController } from './DropdownController';
+import { ActionController } from './ActionController';
+
+jest.useFakeTimers();
+
+describe('DrilldownController', () => {
+  const eventNames = [
+    'w-drilldown:opened',
+    'w-drilldown:closed',
+    'w-drilldown:ready',
+    'w-drilldown:toggled',
+  ];
+
+  const events = {};
+  let application;
+  let errors = [];
+  let controller;
+  let actionElement;
```

Comment:
I would avoid setting controller/actionelement as globals, instead pull them in as needed in each unit test.

It's not ideal to be testing directly against the `controller` instance, this can cause brittle unit tests that rely on the inner implementation of the controller and not the events/DOM that should be used.

I would encourage you to try to review your tests and instead of doing things like `controller.close()` set an element action that will call this instead `data-action="w-dropdown:hidden->w-drilldown#delayedClose"`.